### DISCUSSION TITLE
How to set the scale of the x-axis in a Plot

### DISCUSSION THREAD

**Discussion by jjimenezroda** [emilk/egui#4402] [Q&A] [answered]
Dear all,

I know this is a quite silly question, but I could not really find the way to do this so far. I have the plot show in the following picture:

![image](https://github.com/emilk/egui/assets/12435936/f9e86b26-90dd-4497-aac8-6ea630737e84)

As you can see, it tries to fit the whole plot in the view area (what I mean is that it shows the firts and the last x-value).
Like this I can hardly see anything as this might expand several days (I'm trying to plot time-series).
What I would like is that, when I load the plot, it already looks like this:

![image](https://github.com/emilk/egui/assets/12435936/bcd9a0c0-cdd1-4aea-884f-c936a8148a09)

I manage to get there by zooming only the x-axis but I have no clue how to do this programatically so that the plot renders already like this in the first frame (and later I let the user zoom in or out).

I would really appreciate some light (which I expect to be very easy but I don't manage to find myself).

Thanks a lot upfront!

**Comment by MaeveMcT** [comment]
There is [set_plot_bounds](https://docs.rs/egui_plot/latest/egui_plot/struct.PlotUi.html#method.set_plot_bounds). 
You can set the bounds you'd like for your X axis, but if you want the keep the Y axis scalling automatically, you need [set_auto_bounds](https://docs.rs/egui_plot/latest/egui_plot/struct.PlotUi.html#method.set_auto_bounds).
```
            let min_x_timestamp = ...;
            let max_x_timestamp = ...;
            egui_plot::Plot::new("plot")
                .show(ui, |plot_ui| {
                    plot_ui.set_plot_bounds(PlotBounds::from_min_max(
                        [min_x_timestamp, 0.0],
                        [max_x_timestamp, 0.0],
                    ));
                    plot_ui.set_auto_bounds(Vec2b::new(false, true)); // Let the Y axis auto scale, but not the X axis
                });

```

**Reply by jjimenezroda** [reply]
Thanks a lot for the answer.

It really solves my question, so. I think the answer is correct. For my concrete case it still does not work because it fixes the plot bounds and I want to still let the user scroll and zoom on the x axis. With your proposal the x axis gets locked (or somehow I would need to capture the scroll and zoom values and calculate again the bounds for each update of the screen.

**Reply by MaeveMcT** [reply, marked as answer]
You could set the plot bounds on first draw so the initial view is how you want it to be, but the user can then zoom/scroll as normal. Would this be closer to the behaviour you'd like?
``` 
            
            let mut first_draw = true; // Set somewhere outside of the egui loop
            
            let min_x_timestamp = ...;
            let max_x_timestamp = ...;
            egui_plot::Plot::new("plot")
                .show(ui, |plot_ui| {
                    if first_draw {
                        // Initial plot bound is only set once, rather than every time the plot gets redrawn
                        first_draw = false;
                        plot_ui.set_plot_bounds(PlotBounds::from_min_max(
                            [min_x_timestamp, 0.0],
                            [max_x_timestamp, 0.0],
                        ));
                        plot_ui.set_auto_bounds(Vec2b::new(false, true)); // Let the Y axis auto scale, but not the X axis
                    }
                });
```
